### ISSUE TITLE
refactor: rename Context => Probe

### ISSUE THREAD

**Issue by bassosimone** [ooni/probe-cli#45]
To be done after #44

**Comment by hellais** [comment]
I think that as part of this work, we should also look into simplifying the `Controller` related logic, probably ditching it entirely in favour of something simpler and also looking into using a cancellable `context.Context` for handling running tests, as mentioned by @bassosimone in his review of: https://github.com/ooni/probe-cli/pull/84.

**Comment by bassosimone** [comment]
> I think that as part of this work, we should also look into simplifying the Controller related logic, probably ditching it entirely in favour of something simpler and also looking into using a cancellable context.Context for handling running tests

Yes, good idea. I guess some simplifications will be easier for tests written in Go, but this needs to be verified by taking the time to look into the code and reasoning about that more.
